jupyterhub/templates/NOTES.txt

```diff
@@ -1,26 +1,56 @@
 Thank you for installing JupyterHub!
 
-Your release is named {{ .Release.Name }} and installed into the namespace {{ .Release.Namespace }}.
+Your release is named "{{ .Release.Name }}" and installed into the namespace "{{ .Release.Namespace }}".
 
-You can find if the hub and proxy is ready by doing:
+You can check whether the hub and proxy are ready by running:
 
  kubectl --namespace={{ .Release.Namespace }} get pod
 
-and watching for both those pods to be in status 'Running'.
+and watching for both those pods to be in status 'Running'.{{ println }}
 
-You can find the public IP of the JupyterHub by doing:
+{{- if eq .Values.proxy.service.type "LoadBalancer" }}
+You can find the public (load-balancer) IP of JupyterHub by running:
 
- kubectl --namespace={{ .Release.Namespace }} get svc {{ include "jupyterhub.proxy-public.fullname" . }}
+  kubectl -n {{ .Release.Namespace }} get svc {{ include "jupyterhub.proxy-public.fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[].ip}'
 
 It might take a few minutes for it to appear!
+{{- end }}
+{{- if eq .Values.proxy.service.type "ClusterIP" }}
+You can find the internal (cluster) IP of JupyterHub by running:
 
-Note that this is still an alpha release! If you have questions, feel free to
-  1. Read the guide at https://z2jh.jupyter.org
-  2. Chat with us at https://gitter.im/jupyterhub/jupyterhub
-  3. File issues at https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues
+  kubectl get -n {{ .Release.Namespace }} svc {{ include "jupyterhub.proxy-public.fullname" . }} -o jsonpath='{.spec.clusterIP}'
+{{- end }}
+{{- if eq .Values.proxy.service.type "NodePort" }}
+You can find the NodePorts of JupyterHub by running:
+
+  kubectl --namespace={{ .Release.Namespace }} get svc {{ include "jupyterhub.proxy-public.fullname" . }} -o jsonpath='{range .spec.ports[*]} {.name}: {.port}{"\n"} {end}'
+{{- end }}
+
+{{- if .Values.ingress.enabled }}
 
+You should be able to access JupyterHub using your configured ingress at:
+{{ range $host := .Values.ingress.hosts }}
+  http://{{ $host }}{{ $.Values.hub.baseUrl | trimSuffix "/" }}/
+{{- end }}
+{{- range $tls := .Values.ingress.tls }}
+  {{- range $host := $tls.hosts }}
+  https://{{ $host }}{{ $.Values.hub.baseUrl | trimSuffix "/" }}/
+  {{- end }}
+{{- end }}
+
+{{- end }}
+
+To get full information about the JupyterHub proxy service run:
+
+  kubectl --namespace={{ .Release.Namespace }} get svc {{ include "jupyterhub.proxy-public.fullname" . }}
+
+If you have questions, please:
+
+  1. Read the guide at https://z2jh.jupyter.org
+  2. Ask for help or chat to us on https://discourse.jupyter.org/
+  3. If you find a bug please report it at https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues{{ println }}
 
-{{- if hasKey .Values.singleuser.cloudMetadata "enabled" }}{{ println }}
+{{- if hasKey .Values.singleuser.cloudMetadata "enabled" }}
```

Comment:
Assuming someone would copy paste the deprecation logic, they would need `{{ println }}` in it at least assuming two deprecation messages would trigger - in order to avoid no new blank line between them.

Reply:
Whenever there is a `fail` it is not relevant with later messages as it would stop the rendering, but if it is a warning - multiple warnings can trigger.

Below there is a deprecation warning like this that includes println still while this doesn't any more. I suggest we consistently use println within each.